book/invalidation.md

```diff
@@ -25,25 +28,35 @@ tree_. And while layout-inducing animations are best avoided, lots of
 other interactions require changing the layout tree yet need to be as
 smooth as possible.
 
-One good example is editing text. People type pretty quickly, so even
-a few frames' delay is distracting. But editing changes the HTML tree
-and therefore requires changing the layout tree to reflect new text.
-Currently, that forces our browser to rebuild the layout tree from
-scratch, which can drop multiple frames on complex pages like this
-one. Try, for example, typing into this input box in our toy browser:
+One good example is editing text. People type pretty quickly, so even a few
+frames' delay is distracting. But editing changes the HTML tree and therefore
+requires changing the layout tree to reflect new text. Currently, that forces
+our browser to rebuild the layout tree from scratch, which can drop multiple
+frames on complex pages like this one. Try, for example, loading this chapter
+in our toy browser and typing into this input box:
 
 <input style="width:100%"/>
 
-Typing into an `input` element could be special-cased,[^no-resize] but
+You'll find that it is *much* slower than in a real browser.
+
+Typing into an `input` element could be special-case optimized,[^no-resize] but
 there are other text editing APIs that can't be. For example, the
-`contenteditable` attribute makes any element editable:
+`contenteditable` attribute makes any element editable:[^amazing-ce]
 
 [^no-resize]: The `input` element doesn't change size as you type, and
     the text in the `input` element doesn't get its own layout object,
     so the layout tree is actually the same after each edit.
 
+[^amazing-ce]: The contenteditable attribute is pretty amazing, since
+you can set it on any element on any web site, and thereby turn it into
+a living document. In fact, this is how we implemented the "typo"
+feature for this book(type `Ctrl-E` (`Cmd-E` on a Mac) to turn it on if
+you haven't tried it yet). The source code is [here](/feedback.js); see
+the `typo_mode` function.
+
 ::: {.demo contenteditable=true}
-Click on this <i>formatted</i> <b>text</b> to edit it.
+Click on this <i>formatted</i> <b>text</b> to edit it. It even supports
+rich text edits!
```

Comment:
Does this keep it to one line? I think we want this to be just a little under one line in our toy browser so that people can test invalidation when it wraps to two lines.

Reply:
Fixed